Tools/cases_generator/test_generator.py

```diff
@@ -476,3 +475,26 @@ def test_register():
         }
     """
     run_cases_test(input, output)
+
+def test_cond_effect():
+    input = """
+        inst(OP, (aa, input if (oparg & 1), cc -- xx, output if (oparg & 2), zz)) {
+            output = spam(oparg, input);
+        }
+    """
+    output = """
+        TARGET(OP) {
+            PyObject *cc = POP();
+            PyObject *input = (oparg & 1) ? POP() : NULL;
+            PyObject *aa = PEEK(1);
+            PyObject *xx;
+            PyObject *output;
+            PyObject *zz;
+            output = spam(oparg, input);
+            POKE(1, xx);
+            if (oparg & 2) { PUSH(output); }
+            PUSH(zz);
```

Comment:
Could we stick with PEEK and POKE and here have something like this?

```suggestion
            POKE((oparg & 2) + 2, xx);
            if (oparg & 2) { POKE((oparg & 2) + 1, output); }
            POKE(1, zz);
```

Reply:
Interesting. This would favor cases where the conditional is at the bottom of the inputs (e.g. `a if (...), b, c`). But I think it would become quite horrible if you have several conditions, like we have in MAKE_FUNCTION.

Reply:
If there are several conditions then the concatenation of the conditions (and the unconditional 1's) gives you the index. 
The generator will be simpler, the generated code will have these expressions for indices, but I think that's probably easier to follow than a mixture of pushes and pokes.

Reply:
I'll play around with that.

Reply:
Have a look. The generator sure is a lot simpler, I think I can live with the more complicated generated code. What made this solution so nice is that we already had the symbolic size -- the condition just gets added into that. Thanks for the nudge!